gcore/overview.cpp

```diff
@@ -4361,7 +4404,8 @@ CPLErr GDALRegenerateOverviewsEx(GDALRasterBandH hSrcBand, int nOverviewCount,
 
     const GDALDataType eSrcDataType = poSrcBand->GetRasterDataType();
     const GDALDataType eWrkDataType =
-        GDALDataTypeIsComplex(eSrcDataType)
+        (GDALDataTypeIsComplex(eSrcDataType) &&
+         !STARTS_WITH_CI(pszResampling, "NEAR"))
```

Comment:
Should this also include `&& !STARTS_WITH_CI(pszResampling, "MODE")` ?

Reply:
Yes, but not only... Extra commit to make mode resampling exact too